### PR TITLE
Support TLS for lakeFS

### LINKED ISSUE #5321: Support serve HTTPS without external SSL termination
Enabling TLS support for lakeFS

### PR DESCRIPTION
**Pull request by nopcoder**
Close #5321

### CODE REVIEW

**Comment by Jonathan-Rosenberg** [comment]
@nopcoder I think we should add documentation on how to configure it...

**Comment by nopcoder** [comment]
> @nopcoder I think we should add documentation on how to configure it...

it is best practice to have termination when possible

File: docs/deploy/onprem.md [hunk 1]
```diff
@@ -74,9 +74,20 @@ Connect to your host using SSH:
    ⚠️ Notice that the lakeFS Blockstore type is set to `s3` - This configuration works with S3-compatible storage engines such as [MinIO](https://min.io/){: target="blank" }.
    {: .note }
 
+1. In the case the TLS/SSL termination is not done by your infrastrure's load balancer or cluster manager.
+   lakeFS can be configure to listen and serve HTTPS by adding the following to the configuration:
```
Comment (Jonathan-Rosenberg) [inline docs/deploy/onprem.md:78]:
```suggestion
1. Whenever TLS/SSL is not terminated by your cluster manager or load balancer, add the following configuration to lakeFS to listen and serve HTTPS:
```
maybe...

File: docs/deploy/onprem.md [hunk 2]
```diff
@@ -74,9 +74,20 @@ Connect to your host using SSH:
...
+
+   ```yaml
+   tls:
+     enabled: true
+     cert_file: server.crt   # provide path to your certificate file
+     key_file: server.key    # provide paht to your server private key
```
Comment (Jonathan-Rosenberg) [inline docs/deploy/onprem.md:84]:
```suggestion
     key_file: server.key    # provide path to your server private key
```

**Review by Jonathan-Rosenberg — APPROVED** [review]
COOL

File: docs/deploy/onprem.md [hunk 3]
```diff
@@ -164,16 +165,30 @@ To install lakeFS with Helm:
...
    {: .note }
-   
+
 </div>
 </div>
 
+## Secure connection
+
+It is recommended that TLS/SSL termination by your infrastructure's load balancer or cluster manager.
+
```
Comment (N-o-Z) [inline docs/deploy/onprem.md:181]: I think this needs rephrasing

File: docs/deploy/onprem.md [hunk 4]
```diff
@@ -164,16 +165,30 @@ To install lakeFS with Helm:
...
+To configure your local lakeFS to listen and serve with HTTPS, update `config.yaml` with the additional `tls` section:
```
Comment (N-o-Z) [inline docs/deploy/onprem.md:182]:
```suggestion
To configure your local lakeFS to listen and serve with HTTPS, update the lakeFS config yaml with the additional `tls` section:
```

Comment (nopcoder) [reply docs/deploy/onprem.md:182]: we don't explain on each page how we translate the config.yaml entries from/to env vars names. In this page the guide was based on config.yaml so I've explained how to update the file to include the relevant settings.

File: docs/deploy/onprem.md (hunk 3 above)
Comment (nopcoder) [reply docs/deploy/onprem.md]: updated

**Review by N-o-Z — APPROVED** [review]
LGTM!

**Pull request merged**